### PR TITLE
2152

### PR DESCRIPTION
**Pull request by GSuma21**

### CODE REVIEW

File: src/app/shared/components/sessioncard/sessioncard.component.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        <img [src]="cardData.image" alt="card data image" width="180px" height="auto">
+    </div>
+    <div class="cardDetails">
+
+        <p class="cardDetailBold">{{cardData.title}}</p>
+
+        <p class="cardDetailBold">{{cardData.mentor_Name}}</p>
+        <div class="cardSessionDate">
+            <mat-icon>event</mat-icon>
+            <p>Session date: {{cardData.session_date}}</p>
```
Comment (vishwanath1004) [inline src/app/shared/components/sessioncard/sessioncard.component.html:12]: Use date pipe for the date format

File: src/app/shared/components/sessioncard/sessioncard.component.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+        </div>
+        <div class="cardSessionTime">
+            <mat-icon>alarm</mat-icon>
+            <p>Session time:{{cardData.session_time}}</p>
```
Comment (vishwanath1004) [inline src/app/shared/components/sessioncard/sessioncard.component.html:17]: Use pipe to show time from json

File: src/app/shared/components/sessioncard/sessioncard.component.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+export class SessioncardComponent implements OnInit {
+  @Input() cardData: any;
+
+  constructor(private router: Router) { }
+
+  ngOnInit(): void {
+  }
+
+  redirectToviewDetails() {
+    this.router.navigate([''])
```
Comment (vishwanath1004) [inline src/app/shared/components/sessioncard/sessioncard.component.ts:17]: comment this one for time being

File: src/app/shared/components/sessioncard/sessioncard.component.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+})
+export class SessioncardComponent implements OnInit {
+  @Input() cardData: any;
+
+  constructor(private router: Router) { }
+
+  ngOnInit(): void {
+  }
+
+  redirectToviewDetails() {
```
Comment (vishwanath1004) [inline src/app/shared/components/sessioncard/sessioncard.component.ts:16]: Camelcase follow naming convention

File: src/app/modules/home/components/home/home.component.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-home',
+  templateUrl: './home.component.html',
+  styleUrls: ['./home.component.scss']
+})
+export class HomeComponent implements OnInit {
+  cardDetails = [{
```
Comment (vishwanath1004) [inline src/app/modules/home/components/home/home.component.ts:9]: change the json

File: src/app/shared/components/sessioncard/sessioncard.component.html [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        <img [src]="cardData.image" alt="card data image" width="180px" height="auto">
+    </div>
+    <div class="cardDetails">
+
+        <p class="cardDetailBold">{{cardData.title}}</p>
+
+        <p class="cardDetailBold">{{cardData.mentorName}}</p>
+        <div class="cardSessionDate">
+            <mat-icon>event</mat-icon>
+            <p>Session date: {{cardData.startDate | date}}</p>
```
Comment (vishwanath1004) [inline src/app/shared/components/sessioncard/sessioncard.component.html:12]: you have to specify the date format

File: src/app/modules/home/components/home/home.component.html [hunk 7]
```diff
@@ -0,0 +1,14 @@
+
```
Comment (kiranharidas187) [inline src/app/modules/home/components/home/home.component.html:1]: put a wrapper div

File: src/app/modules/home/components/home/home.component.html [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+<h3 class="allSection">All Sessions</h3>
+
+<div class="categories">
+  <ng-container *ngFor="let item of cardDetails;index as i;">
```
Comment (kiranharidas187) [inline src/app/modules/home/components/home/home.component.html:5]: read on **first** and **last** for ngFor

File: src/app/modules/home/components/home/home.component.html [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+    <div *ngIf="i < last">
+      <app-session-card class="card" [cardData]="item"></app-session-card>
```
Comment (kiranharidas187) [inline src/app/modules/home/components/home/home.component.html:7]: rm class from here

File: src/app/shared/components/sessioncard/sessioncard.component.html [hunk 10]
```diff
@@ -0,0 +1,27 @@
+<mat-card class="cardComponent">
```
Comment (kiranharidas187) [inline src/app/shared/components/sessioncard/sessioncard.component.html]: card-container

File: src/app/shared/components/session-card/session-card.component.scss [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+.cardSessionDate,
+.cardSessionTime {
+    display: flex;
+    flex-direction: row;
+    gap: 10px;
+    font-size: 12px;
+}
+
+.cardDetailBold {
+    font-weight: 550;
```
Comment (kiranharidas187) [inline src/app/shared/components/session-card/session-card.component.scss:67]: correct it

File: src/app/shared/components/session-card/session-card.component.html [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+        </div>
+        <div class="cardSessionTime">
+            <mat-icon>alarm</mat-icon>
+            <p>Session time:{{cardData.startDateUtc | date:'shortTime'}}</p>
+        </div>
+
+        <div class="subCardDetails">
+            <p class="subCardDetailsViewDetails">view details</p>
+            <button class="subCardDetailsButton" mat-raised-button>{{cardData.status | titlecase}}</button>
```
Comment (kiranharidas187) [inline src/app/shared/components/session-card/session-card.component.html:22]: check the material button which has background

File: src/app/shared/shared.module.ts [hunk 13]
```diff
@@ -2,23 +2,31 @@ import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
 import { MatButtonModule } from '@angular/material/button';
 import { MatInputModule } from '@angular/material/input';
-import { DynamicFormComponent } from './components';
+import {MatCardModule} from '@angular/material/card';
+import {MatIconModule} from '@angular/material/icon'
+import { DynamicFormComponent } from './components/dynamic-form/dynamic-form.component';
 import { ReactiveFormsModule } from '@angular/forms';
+import { SessionCardComponent } from './components/session-card/session-card.component';
```
Comment (kiranharidas187) [inline src/app/shared/shared.module.ts:9]: import it from index.ts

**Pull request merged**